packages/uhk-web/src/app/store/reducers/user-configuration.ts

```diff
@@ -27,7 +32,19 @@ export function reducer(state = initialState, action: Action & { payload?: any }
             newKeymap.name = generateName(state.keymaps, newKeymap.name);
             newKeymap.isDefault = (state.keymaps.length === 0);
 
-            changedUserConfiguration.keymaps = state.keymaps.concat(newKeymap);
+            const existingKeymap: Keymap = action.payload;
+
+            const index = sortedIndexBy(
```

Comment:
nice solution, but not linear. I think the linear solution is.

```javascript
const keymaps = [];
let added = false;

for (const keymap of state.keymaps) {
    if (!added && keymap.name.localeCompare(newKeymap.name) > 0) {
        keymaps.push(newKeymap);
        added = true;
     }

    keymaps.push(keymap);
}

changedUserConfiguration.keymaps = keymaps;
```

+1 variable, but 1 loop

Reply:
My solution is linear, but I can see your point. I am goint to change it.